### DISCUSSION TITLE
Using biometric with firebase auth

### DISCUSSION THREAD

**Discussion by Onibenjo** [invertase/react-native-firebase#4727] [Q&A] [answered]
Please is there a way to use fingerprint to authenticate a user on app resume. 

I'm building an app and the user data is quite private.
Currently, clicking the app just automatically authenticates the user since the firebase ask store the credentials. Is there a way to connect this to ask for biometrics?

**Comment by andersonaddo** [comment, marked as answer]
That's not directly supported by firebase; yotur gonna have to do your own hacking and research to get that to work. 

A place to start:
https://stackoverflow.com/questions/39245256/fingerprint-authentication-with-firebase

**Reply by mikehardy** [reply]
There is a react-native fingerprint scanner module, so you can request biometrics based on AppState or on startup. There is no reliable way to detect an app is stopping on mobile as the app can just crash or be killed or the device can power off abruptly.
